### ISSUE TITLE
SageAttention3 in LLM

### ISSUE THREAD

**Issue by taishan1994** [thu-ml/SageAttention#238]
Thanks for your work.

I integrated SageAttention3 into an LLM, but the output results are abnormal. 

Have you tried integrating SageAttention3 into an LLM and checked the results?

In addition, we conducted the following tests and found that the error was relatively large.

```python
import time
import torch
from sageattn import sageattn_blackwell
import torch.nn.functional as F
# from flash_attn_interface import flash_attn_func

is_causal = False

batch_size, head_num, seq_len, head_dim = 64, 16, 8192, 64

q = torch.rand((batch_size, head_num, seq_len, head_dim), dtype=torch.bfloat16, device="cuda:0")
q = q * 6 - 3
k = torch.rand_like(q)
v = torch.rand_like(q)
# q_flash = torch.transpose(q, 1, 2)
# k_flash = torch.transpose(k, 1, 2)
# v_flash = torch.transpose(v, 1, 2)

for _ in range(3):
    o_sage = sageattn_blackwell(q, k, v, is_causal=is_causal)
    o_default = F.scaled_dot_product_attention(q, k, v, is_causal=is_causal)

atol, rtol = 1e-6, 1e-4
equal = torch.allclose(o_sage, o_default, atol=atol, rtol=rtol)
print(f"SageAttention output are approximately equal to Default: {equal}, err_thresh: {atol} (atol), {rtol} (rtol)")
print("Max abs diff:", torch.max(torch.abs(o_sage - o_default)))
print("Mean abs diff:", torch.mean(torch.abs(o_sage - o_default)))
```

<img width="1110" height="538" alt="Image" src="https://github.com/user-attachments/assets/566f328f-5f88-49c9-91d9-a3ee4089f156" />

**Comment by Hprairie** [comment]
Where you ever able to get stable results with SA3 and causal masking, I have noticed that ppl is incredibly bad for SageAttention3 with tasks like wikitext2.
